internal/btf/types.go

```diff
@@ -635,44 +644,34 @@ func alignof(typ Type) (int, error) {
 	}
 }
 
-// Copy a Type recursively.
-func Copy(typ Type) Type {
-	typ, _ = copyType(typ, nil)
-	return typ
-}
-
 // copy a Type recursively.
 //
-// typ may form a cycle.
-//
-// Returns any errors from transform verbatim.
-func copyType(typ Type, transform func(Type) (Type, error)) (Type, error) {
+// typ may form a cycle. If transform is not nil, it is called with the
+// to be copied type, and the return value is copied instead.
+func Copy(typ Type, transform func(Type) Type) Type {
```

Comment:
See my comment on `UnderlyingType` below. `transform` is very flexible and it's not immediately obvious that `UnderlyingType` slots in here.

We can discuss this offline, but let's think through possible use cases for taking a `transform` here on an exported function, and if we could get away with not having it for keeping things simple.